src/service/notification_service/email_sendgrid.rs

```diff
@@ -0,0 +1,142 @@
+use super::{email::EmailMessage, NotificationEmailTransportApi, Result};
+use async_trait::async_trait;
+use rocket::serde::json;
+use serde::Serialize;
+use serde_json::Value;
+
+#[derive(Debug, Clone)]
+pub struct SendgridConfig {
+    api_key: String,
+    url: String,
+}
+
+pub struct SendgridTransport {
+    config: SendgridConfig,
+    client: reqwest::Client,
+}
+
+impl SendgridTransport {
+    /// Creates a new instance of the SendgridTransport.
+    #[allow(dead_code)]
+    pub fn new(config: &SendgridConfig) -> Self {
+        let client = reqwest::Client::new();
+        Self {
+            config: config.to_owned(),
+            client,
+        }
+    }
+
+    async fn send_http_request(&self, message: SendgridMessage) -> Result<()> {
+        let url = format!("{}/v3/mail/send", self.config.url);
+        let request = self
+            .client
+            .post(url)
+            .json(&message)
+            .bearer_auth(&self.config.api_key)
+            .header("Content-Type", "application/json");
```

Comment:
I think `.json` already sets the correct content-type (https://docs.rs/reqwest/latest/src/reqwest/async_impl/request.rs.html#441-459)